movie-app/src/shared-components/switchers-buttons/SwitchersButtons.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState, useCallback } from "react";
+import SwitchersButtonWrapper from "./SwitchersButtonWrapper";
+
+const SwitchersButtons = (props: any) => {
+  const [buttons, updateSearchButtons] = useState(props.searchButtonsParams);
+
+  const activateButton = useCallback(
+    (index: number) => {
+      const arr = buttons.map((item: any, i: number) => {
+        i === index ? (item.active = true) : (item.active = false);
+        return item;
+      });
+      updateSearchButtons(arr);
+    },
+    [buttons]
+  );
+
+  return (
+    <div className="search-buttons-container">
+      <span>{props.switchersTitle}</span>
+      {buttons.map((button: any, index: number) => {
+        return (
+          <SwitchersButtonWrapper
+            onClick={() => activateButton(index)}
```

Comment:
[OPTIONAL] You can avoid the creation of new function on each render by adding `data-` attributes to your component, and calling handler with appropriate arguments based on `e.currentTarget.dataset` attribute value. This helps to avoid unnecessary rerenders